client/src/components/messages/Conversation.tsx

```diff
@@ -53,52 +53,58 @@ const Conversation: React.FC<PropsType> =
   }
   const [label, setLabel] = useState(generateConversationLabel(con));
 
-  const getUnreadMsgsTotal = (conversationId: number, userId: number) => {
-    axios
-      .get(`/api/messages/unread/${conversationId}/${userId}`)
-      .then(({ data }): void => {
-        setUnreadMsgsTotal(data);
-      })
+  const getUnreadMsgsTotal = (conversationId: number, userId: number): void=> {
+    if (con.id) {
+      axios
+        .get(`/api/messages/unread/${conversationId}/${userId}`)
+        .then(({ data }): void => {
+          setUnreadMsgsTotal(data);
+        })
+        .catch((err) => {
+          console.error(`Failed to get unread message total for conversation ${con.id}:`, err);
+        })
+    }
   }
+  // get unread message total on initial render
+  getUnreadMsgsTotal(con.id, user.id);
 
   const markAllMsgsRead = (userId: number, conId: number): void => {
     axios
       .patch(`/api/users/read/${userId}/${conId}`)
       .then(() => {
-        setUnreadMsgsTotal(0)
+        // setUnreadMsgsTotal(0);
         // send socket event to change inbox notification badge
         socket.emit('read-message', {})
       })
+      .then(() => {
+        getUnreadMsgsTotal(conId, userId);
+      })
       .catch((err) => {
         console.error('Failed to mark messages read:\n', err);
       })
   }
 
   socket.on('message', (message) => {
```

Comment:
Not done with this event handler, I'm working to get the notification badges working consistently and to my expectations.